libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Functions/MissingPropertiesFunction.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using AdaptiveExpressions;
+using AdaptiveExpressions.Memory;
+using Microsoft.Bot.Builder.Dialogs.Adaptive;
+using Microsoft.Bot.Builder.Dialogs.Adaptive.Generators;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Functions
+{
+    /// <summary>
+    /// Defines missingProperties(templateName) expression function.
+    /// </summary>
+    /// <remarks>
+    /// This expression will get all variables the template contains.
+    /// </remarks>
+    public class MissingPropertiesFunction : ExpressionEvaluator
+    {
+        /// <summary>
+        /// Function identifier name.
+        /// </summary>
+        public const string Name = "missingProperties";
+
+        private static DialogContext dialogContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MissingPropertiesFunction"/> class.
+        /// </summary>
+        /// <param name="context">Dialog context.</param>
+        public MissingPropertiesFunction(DialogContext context)
+            : base(Name, Function, ReturnType.Array, FunctionUtils.ValidateUnaryString)
+        {
+            dialogContext = context;
+        }
```

Comment:
This is incorrect.  It assumes the dialogContext is immutable and global--neither of which is true.